python/ray/data/_internal/boundaries.py

```diff
@@ -0,0 +1,29 @@
+from typing import Dict, Union
+
+import numpy as np
+
+
+def get_key_boundaries(
+    keys: Union[np.ndarray, Dict[str, np.ndarray]], include_first: bool = True
```

Comment:
Please check my other comments -- these are not really keys, this is rather a projected block

Reply:
Also, do we really need to accept the ndarray?

Reply:
renamed.

For the type, `BlockAccessor.to_numpy()` could return a ndarray or a dict of them.